src/endf_mt.h

```diff
@@ -0,0 +1,222 @@
+      /// lo = 2
+      std::vector<double> lambdas; ///< decay constant for the ith precursor
+
+    } mt460_mf1;
+
+    /// read delayed photon data from a filestream
+    /// \param infile An ENDF filestream
+    mt460_mf1 read_mt460_mf1(std::ifstream &infile);
+
+
+    /// data structure for ENDF decay data
+    typedef struct mt457_mf8 : mt_base {
+      int lis; ///< state of the original nuclide
+      int liso; ///< isomeric state of the original nuclide
+      int nst; ///< nucleus stability flag, 0 = radioactive
+      int nsp; ///< total number of radiation types for which information is given
+      std::vector<std::pair<double,double> > erel; ///< energ, released
+      std::vector<double> styp; ///< Decay radiation
+      std::vector<int> lcon; ///< Continuum spectrum flag
+      std::vector<int> ner; ///< number of tabulated discrete energies
+      std::vector<int> nd; ///< number of decay modes given 
+      std::vector<std::pair<double,double> > fd; ///< discrete spectrum normalization factor
+      std::vector<std::pair<double,double> > eav; ///< averag, decay energy of radiation
+      std::vector<std::pair<double,double> > fc; ///< continuum spectrum normalization
+      std::vector<std::pair<double,double> > er; ///< discrete energ, of radiation
+      std::vector<double> rtyp; ///< decay mode of the nuclide
+      std::vector<double> type; ///< type of beta/ec transition
+      std::vector<std::pair<double,double> > ri; ///< intensity of discrete radiation
+      std::vector<std::pair<double,double> > ris; ///< internal pair formation coefficient
+      std::vector<std::pair<double,double> > ricc; ///< total internal conversion coefficient
+      std::vector<std::pair<double,double> > rick; ///< k-shell internal conversion coefficient
+      std::vector<std::pair<double,double> > ricl; ///< l-shell internal conversion coefficient
+    } mt457_mf8;
+
+    /// read decay data from a filestream
+    /// \param infile An ENDF filestream
+    mt457_mf8 read_mt457_mf8(std::ifstream &infile);
+
+    /// data structure for cross section data
+    typedef struct mf3 : mt_base {
+      std::vector<int> nbt; ///< list of interpolation segments
+      std::vector<int> intn; ///< list of interpolations to be used
+      std::vector<double> energy; ///< ith energy
+      std::vector<double> sigma; ///< ith cross section value
+    } mf3;
+    /// read cross section data from a filestream
+    /// \param infile An ENDF filestream
+    mf3 read_mf3(std::ifstream &infile);
+    
+    typedef struct mf2 : mt_base {
+      int nis;
+
+      // mega-maps map(pair(quantity, unique_index), data)
+      std::map<std::pair<std::string, std::string>, int> data_i;
+      std::map<std::pair<std::string, std::string>, double> data_d;
+      std::map<std::pair<std::string, std::string>, std::vector<int> > data_vi;
+      std::map<std::pair<std::string, std::string>, std::vector<double> > data_vd;
+    } mf2;
```

Comment:
tuples would be nice because then the unique index could be a tuple with unused indices defaulting to -1 instead of a "_" concatenated string which was my current plan . It would make ordering operations much simpler.